### PR TITLE
[test] [sanity] Sanity check for dlc_major_version label

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

*Description:*
Inserted a test that asserts the presence of a LABEL on every image that has been built, named "dlc_major_version", without which the test fails.

*Tests run:*
Tested locally on laptop with 1 failing image and 1 previously released successful image. Both work as expected.

*DLC image/dockerfile:*
N/A

*Additional context:*
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    ecr_client = boto3.client("ecr", region_name=region)
+
+    image_repository, image_tag = get_repository_and_tag_from_image_uri(image)
+    response = ecr_client.batch_get_image(
+        repositoryName=image_repository,
+        imageIds=[{"imageTag": image_tag}],
+        acceptedMediaTypes=["application/vnd.docker.distribution.manifest.v1+json"],
+    )
+    if not response.get("images"):
+        raise ValueError(
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_dlc_labels.py:19]: why is one of these a value error and the other a key error?

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def test_dlc_major_version_label(image, region):
+    ecr_client = boto3.client("ecr", region_name=region)
+
+    image_repository, image_tag = get_repository_and_tag_from_image_uri(image)
+    response = ecr_client.batch_get_image(
+        repositoryName=image_repository,
+        imageIds=[{"imageTag": image_tag}],
+        acceptedMediaTypes=["application/vnd.docker.distribution.manifest.v1+json"],
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_dlc_labels.py:16]: can you provide more info on why this is needed?

File: test/dlc_tests/sanity/test_dlc_labels.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import json
+
+import boto3
+import pytest
+
+from test.test_utils import get_repository_and_tag_from_image_uri, LOGGER
+
+
+def test_dlc_major_version_label(image, region):
```
Comment (arjkesh) [inline test/dlc_tests/sanity/test_dlc_labels.py:9]: nit: add docstring

also you can add a marker for the integration `@pytest.mark.integration('dlc_major_version_label')` or something

Comment (saimidu) [reply test/dlc_tests/sanity/test_dlc_labels.py:9]: We didn't have markers on other sanity tests, so didn't add one here. Can add.

File: test/dlc_tests/sanity/test_dlc_labels.py (hunk 1 above)
Comment (saimidu) [reply test/dlc_tests/sanity/test_dlc_labels.py:19]: The first is an error to be raised when the "images" were not obtained through the ecr client request. I thought that semantically, it would be a missing value, and hence, ValueError.

The other is a key error because values were obtained through the ecr client request, but the specific attribute we need -- imageManifest -- is missing, hence the KeyError to denote a missing dictionary key.

Please recommend any changes that would better represent the errors.

File: test/dlc_tests/sanity/test_dlc_labels.py (hunk 2 above)
Comment (saimidu) [reply test/dlc_tests/sanity/test_dlc_labels.py:16]: Adding comment.

Basically this is a format in which the response needs to be sent, so that the image LABELS can be found on the manifest.

File: test/dlc_tests/sanity/test_dlc_labels.py (hunk 1 above)
Comment (arjkesh) [reply test/dlc_tests/sanity/test_dlc_labels.py:19]: I think they should both be key errors -- think of it this way, if you didn't use .get, you'd get key errors in both scenarios

**Comment by arjkesh** [comment]
can you update the tests run section?

**Comment by saimidu** [comment]
Done. Thanks @arjkesh
